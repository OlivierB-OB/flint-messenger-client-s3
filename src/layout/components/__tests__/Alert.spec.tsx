import React from 'react';
import { shallow } from 'enzyme';
import { Alert } from '../Alert';

describe('Alert', () => {
  it('should display nothing if status is neither success, nor error', async () => {
    const component = shallow(<Alert status="anything" success="ok" />);
    expect(component).toMatchSnapshot();
  });

  it('should display a success message if status is success', async () => {
    const component = shallow(<Alert status="success" success="ok" error="ko" />);
    expect(component).toMatchSnapshot();
  });

  it('should display an error message if status is error', async () => {
    const component = shallow(<Alert status="error" success="ok" error="ko" />);
    expect(component).toMatchSnapshot();
  });

  it('should display nothing if status is success but no success message have been provided', async () => {
    const component = shallow(<Alert status="success" />);
    expect(component).toMatchSnapshot();
  });

  it('should display a default error message if status is error but no error message have been provided', async () => {
    const component = shallow(<Alert status="error" />);
    expect(component).toMatchSnapshot();
  });
});
