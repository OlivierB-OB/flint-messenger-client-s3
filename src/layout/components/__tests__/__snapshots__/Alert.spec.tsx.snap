// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should display a default error message if status is error but no error message have been provided 1`] = `
<WithStyles(ForwardRef(Alert))
  severity="error"
>
  Something really bad happened...
</WithStyles(ForwardRef(Alert))>
`;

exports[`Alert should display a success message if status is success 1`] = `
<WithStyles(ForwardRef(Alert))
  severity="success"
>
  ok
</WithStyles(ForwardRef(Alert))>
`;

exports[`Alert should display an error message if status is error 1`] = `
<WithStyles(ForwardRef(Alert))
  severity="error"
>
  ko
</WithStyles(ForwardRef(Alert))>
`;

exports[`Alert should display nothing if status is neither success, nor error 1`] = `""`;

exports[`Alert should display nothing if status is success but no success message have been provided 1`] = `""`;
